### PR TITLE
3645 sync user during login

### PR DESCRIPTION
**Pull request by kberkos-public**

### CODE REVIEW

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/AuthenticateApi.java [hunk 1]
```diff
@@ -496,6 +505,17 @@ private void setSubjectAndCookies(HttpServletRequest req, HttpServletResponse re
         if (addSSOCookie) {
             ssoCookieHelper.addSSOCookiesToResponse(subject, req, resp);
         }
+        try {
+            Object loginToken = ThreadIdentityManager.setAppThreadIdentity(subject);
+            WebSecurityContext webSecurityContext = (WebSecurityContext) SRTServletRequestUtils.getPrivateAttribute(req, "SECURITY_CONTEXT");
```
Comment (kaczyns) [inline dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/AuthenticateApi.java]: Seems there should be a constant here for the "SECURITY_CONTEXT" string

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/WebAppSecurityCollaboratorImpl.java [hunk 2]
```diff
@@ -567,8 +576,10 @@ public Object preInvoke(HttpServletRequest req, HttpServletResponse resp, String
                 performSecurityChecks(req, resp, receivedSubject, webSecurityContext);
             }
 
-            extraAuditData.put("HTTP_SERVLET_REQUEST", req);
```
Comment (kaczyns) [inline dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/WebAppSecurityCollaboratorImpl.java]: I was wondering why the extra audit data was deleted as this doesn't seem to have anything to do with the sync to thread changes, but then again I don't know much about these parts so maybe it was necessary =)

**Comment by kberkos-public** [comment]
#build

**Comment by kberkos-public** [comment]
#build

**Comment by kberkos-public** [comment]
#build

**Comment by kberkos-public** [comment]
#build

**Pull request merged**